### PR TITLE
Yoga AUFN test environment

### PR DESCRIPTION
**Pull request by sd109**

### CODE REVIEW

File: etc/kayobe/ansible/ceph-config.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/ansible/ceph-config.yml]: File can be removed

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+# Cheat script for a full deployment.
+# This should be used for testing only.
+
+set -eu
+
+BASE_PATH=~
+KAYOBE_BRANCH=stackhpc/wallaby
+KAYOBE_CONFIG_BRANCH=stackhpc/wallaby
+KAYOBE_ENVIRONMENT=aufn-ceph
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: update for yoga

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+# Activate environment
+pushd $BASE_PATH/src/kayobe-config
+source kayobe-env --environment $KAYOBE_ENVIRONMENT
+
+# Configure host networking (bridge, routes & firewall)
+$KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/configure-local-networking.sh
+
+# Deploy Pulp
+$KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/deploy-pulp.sh
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: Can be dropped

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+# Bootstrap the Ansible control host.
+kayobe control host bootstrap
+
+# Sync package & container repositories.
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-sync.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-publish.yml
+
+# NOTE: Building Ubuntu containers locally for now.
+# if $(which dnf 2>/dev/null >/dev/null); then
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: nit: remove

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-sync.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-publish.yml
+# fi
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: ditto

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+kayobe seed vm provision
+
+# Configure the seed host, and deploy a local registry.
+kayobe seed host configure
+
+# NOTE: Building Ubuntu containers locally for now.
+# if ! $(which dnf 2>/dev/null >/dev/null); then
+#     kayobe seed container image build --push
+#     kayobe overcloud container image build --push
+# fi
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: nit: remove

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+ cd $BASE_PATH/src/kayobe && \
+ ./dev/tenks-deploy-overcloud.sh ./tenks)
+
+# Inspect and provision the overcloud hardware:
+kayobe overcloud inventory discover
+kayobe overcloud hardware inspect
+kayobe overcloud provision
+kayobe overcloud host configure
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/cephadm.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/ceph-config.yml
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: Remove

File: etc/kayobe/environments/aufn-ceph/cephadm.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/cephadm.yml:1]: Use the config from etc/kayobe/environments/ci-multinode/cephadm.yml

File: etc/kayobe/environments/aufn-ceph/deploy-pulp.sh [hunk 9]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/deploy-pulp.sh]: Remove this file

File: etc/kayobe/environments/aufn-ceph/inventory/group_vars/overcloud/ansible-python-interpreter [hunk 10]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/inventory/group_vars/overcloud/ansible-python-interpreter]: File can be removed

File: etc/kayobe/environments/aufn-ceph/inventory/group_vars/seed-hypervisor/ansible-python-interpreter [hunk 11]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/inventory/group_vars/seed-hypervisor/ansible-python-interpreter]: File can be removed

File: etc/kayobe/environments/aufn-ceph/inventory/group_vars/seed/ansible-python-interpreter [hunk 12]
```diff
@@ -0,0 +1,3 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/inventory/group_vars/seed/ansible-python-interpreter]: File can be removed

File: etc/kayobe/environments/aufn-ceph/inventory/groups [hunk 13]
```diff
@@ -0,0 +1,106 @@
+# Kayobe groups inventory file. This file should generally not be modified.
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/inventory/groups:1]: Anything that matches etc/kayobe/inventory/groups can be removed

File: etc/kayobe/environments/aufn-ceph/ipa.yml [hunk 14]
```diff
@@ -0,0 +1,35 @@
+---
+# Ironic Python Agent (IPA) configuration.
+
+###############################################################################
+# Ironic Python Agent (IPA) images configuration.
+
+# Suffix of upstream Ironic deployment image files. Default is based on
+# {{ openstack_branch }}.
+#ipa_images_upstream_url_suffix:
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/ipa.yml]: This and other comments can be removed

File: etc/kayobe/environments/aufn-ceph/kolla.yml [hunk 15]
```diff
@@ -0,0 +1,267 @@
+---
+# Kayobe Kolla configuration.
+
+###############################################################################
+# Kolla-ansible inventory configuration.
+
+# Full custom seed inventory contents.
+#kolla_seed_inventory_custom:
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla.yml]: This and other comments can be removed

File: etc/kayobe/environments/aufn-ceph/kolla/globals.yml [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+# denominator.
+nova_compute_virt_type: qemu
+
+# Reduce the control plane's memory footprint by limiting the number of worker
+# processes to one per-service.
+openstack_service_workers: "1"
+
+# Prevent Docker from manipulating iptables. Docker changes the default policy
+# on the FORWARD chain, which prevents traffic from reaching instances.
+docker_disable_default_iptables_rules: true
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla/globals.yml]: Not required in yoga

File: etc/kayobe/environments/aufn-ceph/stackhpc.yml [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+# Kolla configuration.
+#
+# Docker namespace to use for Kolla images. Default is 'kolla'.
+kolla_docker_namespace: stackhpc-dev
+
+###############################################################################
+# Seed configuration.
+
+# Using a Pulp container on the seed-hypervisor.
+seed_pulp_container_enabled: false
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/stackhpc.yml]: This can be removed

File: etc/kayobe/environments/aufn-ceph/stackhpc.yml [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+# StackHPC configuration.
+
+# Base URL of the StackHPC Test Pulp service.
+stackhpc_release_pulp_url: "http://pulp-server.internal.sms-cloud:8080"
+
+# No client certs required to access Test Pulp.
+stackhpc_release_pulp_client_cert:
+stackhpc_release_pulp_client_key:
+
+pulp_url: http://192.168.33.4:8080 #"http://{{ admin_oc_net_name | net_ip(groups['seed-hypervisor'][0]) }}:80"
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/stackhpc.yml]: This can be removed

File: etc/kayobe/inventory/groups [hunk 19]
```diff
@@ -60,7 +60,7 @@ seed
 controllers
 network
 monitoring
-storage
+#storage
```
Comment (markgoddard) [inline etc/kayobe/inventory/groups]: Revert to default

File: etc/kayobe/environments/aufn-ceph/stackhpc.yml [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+pulp_username: admin
+# NOTE: must match deploy-pulp.sh
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/stackhpc.yml]: Remove comment

File: etc/kayobe/environments/aufn-ceph/kolla.yml [hunk 21]
```diff
@@ -0,0 +1,267 @@
...
+
+# Custom overcloud inventory containing a mapping from services to components.
+kolla_overcloud_inventory_custom_services: "{{ lookup('template', kayobe_env_config_path ~ '/kolla/inventory/overcloud-services.j2') }}"
+
+# Full custom overcloud inventory contents. By default this will be the
+# concatenation of the top level, component, and service inventories.
+#kolla_overcloud_inventory_custom:
+
+# Don't give storage nodes to kolla-ansible - we're using Ceph-ansible.
+kolla_overcloud_inventory_storage_groups: []
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla.yml]: Let's comment this one out

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh (hunk 2 above)
Comment (sd109) [reply etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: What should KAYOBE_ENVIRONMENT be set to here?

**Comment by sd109** [comment]
Are there any more changes that need made before I try testing this?

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh (hunk 2 above)
Comment (markgoddard) [reply etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: ah, environment is the same for now. For your testing you'll need to change the branch to yoga-aufn

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 22]
```diff
@@ -0,0 +1,118 @@
...
+$KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/configure-local-networking.sh
+
+# Bootstrap the Ansible control host.
+kayobe control host bootstrap
+
+# Sync package & container repositories.
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-sync.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-publish.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-sync.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-publish.yml
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: These need to move after the `seed service deploy`

File: etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2 [hunk 23]
```diff
@@ -0,0 +1,561 @@
...
+cinder
+
+[cinder-backup:children]
+storage
+
+[cinder-scheduler:children]
+cinder
+
+[cinder-volume:children]
+cinder
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2:124]: Note that this was changed from the default, and should be kept.

File: etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2 [hunk 24]
```diff
@@ -0,0 +1,561 @@
+# This inventory section provides a mapping of services to components.
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2:1]: This might need an update for yoga. Here is the source: https://github.com/stackhpc/kayobe/blob/stackhpc/yoga/ansible/roles/kolla-ansible/templates/overcloud-services.j2

File: etc/kayobe/environments/aufn-ceph/storage.yml [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+# Storage node LVM configuration.
+
+# List of storage volume groups. See mrlesmithjr.manage-lvm role for
+# format.
+#storage_lvm_groups:
+storage_lvm_groups: []
+
+# Default list of storage volume groups. See mrlesmithjr.manage-lvm role for
+# format.
+#storage_lvm_groups_default:
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/storage.yml]: nit: could remove these

**Review by markgoddard — CHANGES_REQUESTED** [review]
Looking better

**Review by markgoddard — COMMENTED** [review]
LGTM, give it a whirl.

File: etc/kayobe/ansible/cephadm-commands.yml [hunk 26]
```diff
@@ -0,0 +1,11 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/ansible/cephadm-commands.yml]: This should be a separate PR to stackhpc/xena (PR1)

File: etc/kayobe/ansible/cephadm.yml [hunk 27]
```diff
@@ -1,5 +1,6 @@
 ---
 # Deploy Ceph via Cephadm. Create EC profiles, CRUSH rules, pools and keys.
+- import_playbook: cephadm-commands.yml
```
Comment (markgoddard) [inline etc/kayobe/ansible/cephadm.yml]: This should be a separate PR to stackhpc/xena (PR1)

File: etc/kayobe/cephadm.yml [hunk 28]
```diff
@@ -91,7 +91,7 @@ kolla_ceph_services:
 kolla_ceph_cinder_volume_required: "{{ kolla_enable_cinder | bool }}"
 
 # Whether to generate Ceph configuration for Cinder backup.
-kolla_ceph_cinder_backup_required: "{{ kolla_enable_cinder_backup | default(false) | bool }}"
+kolla_ceph_cinder_backup_required: "{{ kolla_enable_cinder_backup | default(true) | bool }}"
```
Comment (markgoddard) [inline etc/kayobe/cephadm.yml]: This should be a separate PR to stackhpc/xena (PR2)

File: etc/kayobe/dnf.yml [hunk 29]
```diff
@@ -41,7 +41,7 @@
 #     file: myrepo
 #     gpgkey: http://gpgkey
 #     gpgcheck: yes
-dnf_custom_repos: "{{ dnf_custom_repos_el8 | combine(lookup('vars', 'dnf_custom_repos_' ~ ansible_facts.distribution | lower )) }}"
+stackhpc_dnf_repos: "{{ dnf_custom_repos_el8 | combine(lookup('vars', 'dnf_custom_repos_' ~ ansible_facts.distribution | lower )) }}"
```
Comment (markgoddard) [inline etc/kayobe/dnf.yml]: This should be in a separate PR to stackhpc/xena (PR3)

File: etc/kayobe/inventory/group_vars/overcloud/stackhpc-dnf-repos [hunk 30]
```diff
@@ -0,0 +1,5 @@
+# Only use local pulp mirrors for overcloud hosts
+# to avoid situations where the seed-hypervisor 
+# tries to use a local pulp repo on the seed VM
+# before the seed vm has been provisioned
+dnf_custom_repos: "{{ stackhpc_dnf_repos }}"
```
Comment (markgoddard) [inline etc/kayobe/inventory/group_vars/overcloud/stackhpc-dnf-repos]: PR3

File: etc/kayobe/kolla/globals.yml [hunk 31]
```diff
@@ -1,7 +1,11 @@
 # yamllint disable-file
 ---
-docker_yum_baseurl: "{{ stackhpc_repo_docker_url }}"
-docker_yum_gpgkey: "https://download.docker.com/linux/centos/gpg"
+
+# To work around issue of trying to install docker from 
+# empty pulp server, use upstream docker dnf repo on 
+# non-overcloud hosts
+enable_docker_repo: {% raw %}"{{ 'overcloud' not in group_names }}"{% endraw %}
```
Comment (markgoddard) [inline etc/kayobe/kolla/globals.yml]: PR3

**Review by markgoddard — CHANGES_REQUESTED** [review]
Extract the non-AUFN bits please

File: etc/kayobe/cephadm.yml (hunk 28 above)
Comment (sd109) [reply etc/kayobe/cephadm.yml]: This change was already made into a [separate PR](https://github.com/stackhpc/stackhpc-kayobe-config/pull/243) and has since been merged into stackhpc/xena and stackhpc/yoga by the look of it

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 32]
```diff
@@ -0,0 +1,169 @@
...
+REMOTE_COMMAND="docker exec pulp sh -c 'echo $PULP_HOST | tee -a /etc/hosts'"
+ssh stack@$SEED_IP $REMOTE_COMMAND
+
+# Sync package & container repositories.
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-sync.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-repo-publish.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-sync.yml
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/pulp-container-publish.yml
+
+kayobe seed container image build bifrost_deploy
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: nit: don't need to specify bifrost_deploy

Comment (markgoddard) [reply etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: Actually, we should be getting this image from pulp, so don't need to build at all

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 33]
```diff
@@ -0,0 +1,169 @@
...
+
+# Re-run full task to set up bifrost_deploy etc. using newly-populated pulp repo
+kayobe seed service deploy
+
+
+# Deploying the seed restarts networking interface, run configure-local-networking.sh again to re-add routes.
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: Do we need this here? We have it on L94

File: etc/kayobe/environments/aufn-ceph/configure-local-networking.sh [hunk 34]
```diff
@@ -0,0 +1,85 @@
...
+# hosting the VM.
+seed_hv_private_ip=$(ip a show dev $iface | awk '$1 == "inet" { gsub(/\/[0-9]*/,"",$2); print $2; exit }')
+
+# Forward the following ports to the controller.
+# 80: Horizon
+# 6080: VNC console
+forwarded_ports="80 6080"
+
+# IP of the seed hypervisor on the OpenStack 'public' network created by init-runonce.sh.
+public_ip="10.0.2.1"
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/configure-local-networking.sh]: I think this doesn't match the current external network. I think we don't need it since we assign the seed hypervisor an IP on that network via kayobe.

File: etc/kayobe/environments/aufn-ceph/configure-openstack.sh [hunk 35]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+BASE_PATH=~
+VENV_DIR=$BASE_PATH/venvs/ansible
+cd $BASE_PATH/src/
+[[ -d openstack-network-config ]] || git clone https://github.com/stackhpc/openstack-config-multinode.git -b geneve openstack-network-config
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/configure-openstack.sh]: Do we still need to use a geneve branch?

File: etc/kayobe/environments/aufn-ceph/init-runonce.sh [hunk 36]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/init-runonce.sh]: I think we can drop this file

File: etc/kayobe/environments/aufn-ceph/inventory/groups [hunk 37]
```diff
@@ -0,0 +1,29 @@
...
+# It declares the top-level groups and sub-groups.
+
+###############################################################################
+# Ceph groups
+
+[storage-ceph]
+storage0
+storage1
+storage2
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/inventory/groups]: We shouldn't really have hosts in the groups file. This should get generated via the group map in overcloud.yml

File: etc/kayobe/environments/aufn-ceph/kolla/config/bifrost/bifrost.yml [hunk 38]
```diff
@@ -0,0 +1,12 @@
...
+# Don't build an IPA deployment image, instead download upstream images.
+create_ipa_image: false
+download_ipa: true
+
+# Use a locally hosted CentOS8 cloud image.
+use_cirros: true
+{% if os_distribution == 'ubuntu' %}
+cirros_deploy_image_upstream_url: "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img"
+{% else %}
+cirros_deploy_image_upstream_url: "https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-GenericCloud-8-20210210.0.x86_64.qcow2"
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla/config/bifrost/bifrost.yml]: We could probably go for the more modern https://cloud.centos.org/centos/8-stream/x86_64/images/CentOS-Stream-GenericCloud-8-20220913.0.x86_64.qcow2

File: etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2 [hunk 39]
```diff
@@ -0,0 +1,561 @@
...
+
+[ironic-neutron-agent:children]
+neutron
+
+# Cinder
+[cinder-api:children]
+cinder
+
+[cinder-backup:children]
+storage
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla/inventory/overcloud-services.j2]: cinder-backup should also move to the cinder group

File: etc/kayobe/environments/aufn-ceph/neutron.yml [hunk 40]
```diff
@@ -0,0 +1,15 @@
+---
+kolla_neutron_ml2_network_vlan_ranges:
+  - physical_network: "physnet1"
+
+kolla_neutron_ml2_type_drivers:
+  - flat
+  - vlan
+  - vxlan
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/neutron.yml]: Should we remove vxlan?

File: etc/kayobe/environments/aufn-ceph/neutron.yml [hunk 41]
```diff
@@ -0,0 +1,15 @@
...
+  - geneve
+
+kolla_neutron_ml2_tenant_network_types:
+  - flat
+  - vlan
+  - vxlan
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/neutron.yml]: ditto

File: etc/kayobe/environments/aufn-ceph/stackhpc.yml [hunk 42]
```diff
@@ -0,0 +1,24 @@
...
+
+###############################################################################
+# StackHPC configuration.
+
+# Base URL of the StackHPC Test Pulp service.
+stackhpc_release_pulp_url: "http://pulp-server.internal.sms-cloud:8080"
+
+# No client certs required to access Test Pulp.
+stackhpc_release_pulp_client_cert:
+stackhpc_release_pulp_client_key:
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/stackhpc.yml]: These variables were removed from the base config

File: etc/kayobe/environments/aufn-ceph/storage.yml [hunk 43]
```diff
@@ -0,0 +1,9 @@
+---
+
+###############################################################################
+# Storage node LVM configuration.
+
+# List of storage volume groups. See mrlesmithjr.manage-lvm role for
+# format.
+#storage_lvm_groups:
+storage_lvm_groups: []
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/storage.yml:11]: Do you remember why this was required?

File: etc/kayobe/seed-hypervisor.yml [hunk 44]
```diff
@@ -5,6 +5,7 @@
 # User with which to access the seed hypervisor via SSH during bootstrap, in
 # order to setup the Kayobe user account. Default is {{ os_distribution }}.
 #seed_hypervisor_bootstrap_user:
+seed_hypervisor_bootstrap_user: "{{ lookup('env', 'USER') }}"
```
Comment (markgoddard) [inline etc/kayobe/seed-hypervisor.yml]: Could you move this variable to the aufn-ceph environment?

**Review by markgoddard — CHANGES_REQUESTED** [review]
Sorry for the slow review. If you rebase your changes onto stackhpc/xena, it should pick up the bits you extracted.

File: etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh [hunk 45]
```diff
@@ -0,0 +1,169 @@
...
+###########################################
+
+# Cheat script for a full deployment.
+# This should be used for testing only.
+
+set -eu
+
+BASE_PATH=~
+KAYOBE_BRANCH=stackhpc/yoga
+KAYOBE_CONFIG_BRANCH=yoga-aufn
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/a-universe-from-nothing.sh]: This should change to stackhpc/yoga before merge

File: etc/kayobe/environments/aufn-ceph/neutron.yml (hunk 40 above)
Comment (sd109) [reply etc/kayobe/environments/aufn-ceph/neutron.yml]: The playbooks within the open stack-network-config repo try to create vxlan networks, so unless we modify that config for this use case then we need to leave vxlan enabled in neutron

File: etc/kayobe/environments/aufn-ceph/neutron.yml (hunk 41 above)
Comment (sd109) [reply etc/kayobe/environments/aufn-ceph/neutron.yml]: See previous comment

File: etc/kayobe/environments/aufn-ceph/neutron.yml (hunk 40 above)
Comment (markgoddard) [reply etc/kayobe/environments/aufn-ceph/neutron.yml]: If we're using OVN as the networking driver then we should create GENEVE tenant networks rather than VXLANs. Is it possible to omit the network type in openstack-config and let neutron choose for you?

File: etc/kayobe/environments/aufn-ceph/configure-openstack.sh [hunk 46]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+BASE_PATH=$1
+VENV_DIR=$BASE_PATH/venvs/ansible
+cd $BASE_PATH/src/
+[[ -d openstack-config ]] || git clone https://github.com/stackhpc/openstack-config-multinode.git -b generic-network openstack-config
```
Comment (sd109) [inline etc/kayobe/environments/aufn-ceph/configure-openstack.sh]: Note: Remember to change checkout branch once [this PR](https://github.com/stackhpc/openstack-config-multinode/pull/2) is merged.

**Comment by sd109** [comment]
I *think* this is now ready to go.

**Review by markgoddard — DISMISSED** [review]
Great work Scott. I rebased the change to remove the commits that were pulled out into separate PRs. I pushed your original branch to yoga-aufn-og in case you wanted to compare.

**Comment by markgoddard** [comment]
Looks like there are some pep8 issues: https://github.com/stackhpc/stackhpc-kayobe-config/actions/runs/3904414510/jobs/6670166003

**Comment by markgoddard** [comment]
Could you add some basic docs in `doc/source/contributor/environments/aufn-ceph.rst`? It can be in a separate PR

**Comment by markgoddard** [comment]
Seems we have some pep8 failures

**Comment by sd109** [comment]
How do I fix the final pep8 issue? It's complaining about {% if ... %} syntax in a yaml file.

Will add some docs separately here: [https://github.com/stackhpc/stackhpc-kayobe-config/tree/yoga-aufn-docs](https://github.com/stackhpc/stackhpc-kayobe-config/tree/yoga-aufn-docs)

File: etc/kayobe/environments/aufn-ceph/kolla/config/bifrost/bifrost.yml [hunk 47]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (markgoddard) [inline etc/kayobe/environments/aufn-ceph/kolla/config/bifrost/bifrost.yml:2]: Strictly this is a Jinja2 file, rather than pure YAML. Sometimes it parses ok, but if you use an if like on L8 it won't be valid YAML.

```suggestion
# yamllint disable-file
---
```

**Review by markgoddard — APPROVED** [review]
Nice work!

**Comment by markgoddard** [comment]
How do you feel about squashing the commits? Are they useful to keep separate?

**Comment by markgoddard** [comment]
> How do you feel about squashing the commits? Are they useful to keep separate?

(Merge pull request is a drop down)
